sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -89,22 +89,22 @@ class KotlinSensor(
             fileSystem.predicates().hasType(InputFile.Type.MAIN)
         )
 
-        val filesToAnalyze = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
+        val filePartition: Pair<Iterable<InputFile>, Iterable<InputFile>> = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
             if (canSkipUnchangedFiles(sensorContext)) {
                 LOG.debug("The Kotlin analyzer is running in a context where it can skip unchanged files.")
                 var totalFiles = 0
-                mainFiles
-                    .filter {
-                        totalFiles++
-                        it.status() != InputFile.Status.SAME
-                    }.also {
-                        LOG.info("Only analyzing ${it.size} changed Kotlin files out of ${totalFiles}.")
-                    }
+                val partition = mainFiles.partition {
+                    totalFiles++
+                    it.status() == InputFile.Status.SAME && reuseCPDTokens(it, sensorContext)
+                }
+                LOG.info("Only analyzing ${partition.second.size} changed Kotlin files out of ${totalFiles}.")
+                partition
```

Comment:
I would recommend using `.also` to log, as this is more explicit about the semantic dependencies in the code. It also removes the need for the expression statement to return `partition`